custom_components/volkswagen_we_connect_id/__init__.py

```diff
@@ -2,9 +2,11 @@
 from __future__ import annotations
 
 from datetime import timedelta
+import functools
 import logging
-import asyncio
+import threading
```

Comment:
This may look alarmingly like I’m replacing `asyncio` with `threading`, but not really! 😀 `asyncio` was deleted simply because it was not being used in this module (even before this PR), and `threading` was added for the use of `threading.Lock`.

By the way, this PR does not cause any threads to be spawn or any new execution parallelism. The threads were already there before this PR, making multiple parallel calls to the WeConnect API. This PR simply adds some thread coordination to prevent unnecessary API calls.